server/core/src/main/scala/org/infinispan/server/core/transport/ExtendedByteBuf.scala

```diff
@@ -25,6 +29,11 @@ object ExtendedByteBuf {
       }
    }
 
+   def readOptRangedBytes(bf: ByteBuf): Option[Array[Byte]] = {
+      val length = readUnsignedInt(bf)
+      if (length < 0) None else Some(readRangedBytes(bf, length))
```

Comment:
Nevermind, found out the reason why. Our VInts work with negative values but always produce 5 bytes encoded values even for small values like ```-1```